client/common/tx/confirmation.go

```diff
@@ -37,11 +35,12 @@ func WaitMined(ctx context.Context, backend Backend, txHash common.Hash) (*types
 		} else {
 			logger.Trace("Transaction not yet mined")
 		}
-		// Wait for the next round.
+
```

Comment:
maybe we could check ctx timeout and break the func being a loop

Reply:
The check is a few lines below the code fragment :)